crates/rolldown_plugin_replace/tests/form/typescript_declare/artifacts.snap

```diff
@@ -21,6 +21,6 @@ console.log(replaced);
 - ../input.ts
 (1:0) "console." --> (2:0) "console."
 (1:8) "log(" --> (2:8) "log("
-(1:12) "NAME)" --> (2:12) "replaced)"
-(1:17) "\n" --> (2:21) ";\n"
+(1:12) "NAME" --> (2:12) "replaced)"
+(1:16) ")\n" --> (2:21) ";\n"
```

Comment:
This minor diff seem to be due to collapse sourcemap and likely the lookup logic here https://github.com/oxc-project/oxc/blob/6a98ef1a0c4e68251078483f57d17ecaebd9c34e/crates/oxc_sourcemap/src/sourcemap.rs#L208-L210, which picks up previous token when exact token is not found.

I can reproduce the same with using magic string in js plugin https://stackblitz.com/edit/github-9uyrrq?file=rolldown.config.mjs

I don't think this is a big issue, so probably we can follow up later. Perhaps we can add it to https://github.com/rolldown/rolldown/issues/2737 just in case.

Reply:
>  which picks up previous token when exact token is not found.

So Is this wrong compared to the actually behavior of looking up source based on sourcemap by chrome?

Reply:
Good question, but not sure the answer. I wasn't indicating sourcemap collapsing is "wrong", but the combination of these steps are affecting the result:
- codegen appends `;` and also add mapping as the next char of `)` which is `\n`,
- then rolldown collapses codegen's map and magic-string's map and it tries to restore the mapping of `;` generated by codegen,
- here previously there was a mapping for `\n`, but it's missing now, so it picks up `)` instead.

For example, if the original code had `;` after the statement, this doesn't happen.

Reply:
> codegen appends ; and also add mapping as the next char of ) which is \n,

Seems wrong to me. If the original source doesn't have `;`, the new generated `;` should point to nothing.

I'll add these to #2737.

Reply:
FYI @Boshen cc